### ISSUE TITLE
Packaging page

### ISSUE THREAD

**Issue by mattagape** [UCL/rsd-engineeringcourse#96]
1. I don't have the tree command on my machine so I'm not sure what it does.

2. One line says:
`from greetings.greeter import greet`
but all we had on the last page was greeter.py; there wasn't a greet() function. Is that why we are seeing tracebacks? If so that seems rather pointless.

3. The greet() function appears further down in the page. Again this is pointless.

4. Need to explain:
`from .greeter import greet # Note python 3 relative import`

5. Need to explain entry points code

In summary I think this page needs a lot of work/rewriting

**Comment by ageorgou** [comment]
1. Distribution tools: Formatting changes - should the line starting "Language-specific tools" be a bullet list?

1. It looks like something is missing: the `greet` package and files must already be installed in order for this to work, but their code is given later. Perhaps first show what needs to be done (which files we need, their structure, code, etc), and then move the code showing how to use the package after that? Or keep the examples where they are, but not as input cells, just code markdown. 

1. Developer install: Change `python setup.py develop` to `pip install -e .`? Also, these sections from here onwards could perhaps be marked as optional/advanced?

**Comment by dpshelio** [comment]
link to software carpentry is wrong.

**Comment by dpshelio** [comment]
From #141 
- [ ] 1. Needed to show the tree directory structure. It's left for now
- [x] 2. It's part of the repository. They are shown in the directory. Tracebacks are to do with some python2 remanents.
- [x] 3. The function `greet()` is shown later in the page to show how to add documentation. I've updated that so it doesn't `writefile` but shows it, and said about the documentation.
- [x] 4. That's explained in the previous notebook now.
- [x] 5. explained now.
- [x] 1. bullet list
- [x] 2. Change some of the bits as `%pycat` instead of writing, to show what's been written.
- [x] 3. Updated to `pip install -e .`
- [x] 1. Link fixed

**Comment by Saransh-cpp** [comment]
The whole page was revamped recently ✅
